### DISCUSSION TITLE
glpi-netdiscovery/injector

### DISCUSSION THREAD

**Discussion by FrancoisG25** [glpi-project/glpi-agent#304] [Q&A]
Good afternoon,

Could you please help me ?

I'm trying to net discovery but everything is  in "Equipment refused by rules log"  : [Global constraint (name)]

My problem is if i remove this rule the import is empty : 
![image](https://user-images.githubusercontent.com/117462288/208943251-9181759c-3c1c-4a47-8578-d6d8fdd513dc.png)

![image](https://user-images.githubusercontent.com/117462288/208944292-1a9c799c-0df9-4a94-b9cf-9393cae4ea7a.png)

He get some informations like the ip adress and others things for exemple why he didn't put what he has then i can add what is messing ?
![image](https://user-images.githubusercontent.com/117462288/208943855-ac7d5c98-1772-49ae-b716-1959728f9d58.png)

![image](https://user-images.githubusercontent.com/117462288/208941699-96e60b37-9b52-4fcd-bf29-0b675670d96f.png)

Any ideas ?

Thanks for your time

**Comment by g-bougard** [comment]
Hi @FrancoisG25 
netdiscovery task is only for network devices and printers supporting snmp. Computer are not supported as we can't catch UUID or serialnumber this way.
If you need to inventory computer remotely, you'll have to use the `remoteinventory` task:
See [Remoteinventory task documentation](https://glpi-agent.readthedocs.io/en/latest/tasks/remote-inventory.html)
Or you also can install glpi-agent service on each computers.

**Reply by FrancoisG25** [reply]
@g-bougard Thanks for your reply.

First i try remote inventory but i have a issue with ssh :
check failure: Can't run simple command on remote, check server is up and ssh access is setup.
(ssh root@ip work)   ssh -l root ip id -u return 0

The other thing as you say, only the switchs and printers come from the netdiscovery.

 Any solution for firewall cause i have the same issue as computer  : Import denied [type]: Computer, [name]
I don't want/can't install client and as i say remoteinventory didn't work even if i find solution by changing the sshd_config don't want to do it on firewall.

Thanks

**Reply by g-bougard** [reply]
For remoteinventory, you should try the [nightly build](https://nightly.glpi-project.org/glpi-agent/) as some fixes has been included for this task.
For your firewall, what kind of device it is ?

**Reply by automation-h24** [reply]
I have the same issue.
My firewall is opnsense.

**Reply by g-bougard** [reply]
Hi @automation-h24 
have you carefully read my answers ? If you're running the task to inventory computers, this is just wrong and should better use remoteinventory task. Otherwise you have another problem and should open another discussion.

**Comment by g-bougard** [comment]
Speaking of remoteinventory task, people should know latest GLPI-Agent version now includes an updated ToolBox plugin which will help you to run this task.
Take a look at this doc: https://glpi-agent.readthedocs.io/en/latest/plugins/toolbox-plugin.html
